### ISSUE TITLE
[postgres] CTE with column names triggers L022

### ISSUE THREAD

**Issue by enote-kane** [sqlfluff/sqlfluff#3474]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

Relates to but not fixed with: #2136 (PR #2139)

### What Happened

An unexpected L022 error was raised.

### Expected Behaviour

No error

### Observed Behaviour

```
L:   6 | P:  13 | L022 | Blank line expected but not found after CTE closing
                       | bracket.
```

### How to reproduce

```
$ sqlfluff lint --nocolor --verbose --dialect postgres 2022-06-21.cte-newline.sql 
==== sqlfluff ====
sqlfluff:                1.0.0 python:                 3.8.10
implementation:        cpython verbosity:                   1
dialect:              postgres templater:               jinja
rules:                                all
==== readout ====

=== [ path: 2022-06-21.cte-newline.sql ] ===

== [2022-06-21.cte-newline.sql] FAIL                                                                                                                                              
L:   6 | P:  13 | L022 | Blank line expected but not found after CTE closing
                       | bracket.
==== summary ====
violations:        1 status:         FAIL
All Finished!
```

Test file `2022-06-21.cte-newline.sql`:
```sql
WITH
cte_1 AS (
    SELECT 1 AS var
),

cte_2 (var) AS (
    SELECT 2
)

SELECT
    cte_1.var,
    cte_2.var
FROM cte_1, cte_2;
```

### Dialect

postgres

### Version

1.0.0

### Configuration

_default_

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
OK I see the issue with the last fix.

This code:

https://github.com/sqlfluff/sqlfluff/blob/17813abdfd31f570ab0950c3197374e00b2113bc/src/sqlfluff/rules/L022.py#L66-L75

Looks for the preceding keyword and checks if it's `AS`. And in this case it IS `AS` due to `SELECT 1 AS var`.

I don't think we should be looking all the way back to the last keyword. I think it's fine to just look back to the first non-whitespace bit of code (whether keyword or not) and then make the decision.

PR coming up...

**Comment by enote-kane** [comment]
@tunetheweb @barrywhart In the latest version 1.1.0 this is still not fixed.

A retest with the exact same file and default configuration:
```
$ sqlfluff lint --nocolor --verbose --dialect postgres 2022-06-21.cte-newline.sql
==== sqlfluff ====
sqlfluff:                1.1.0 python:                 3.8.10
implementation:        cpython verbosity:                   1
dialect:              postgres templater:               jinja
rules:                                all
==== readout ====

=== [ path: 2022-06-21.cte-newline.sql ] ===

== [2022-06-21.cte-newline.sql] FAIL                                                                                                                                              
L:   6 | P:  13 | L022 | Blank line expected but not found after CTE closing
                       | bracket.
==== summary ====
violations:        1 status:         FAIL
All Finished!
```
